src/Exception/EntityNotFoundException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Linio\Common\Exception;
+
+use Throwable;
+
+class EntityNotFoundException extends ClientException
+{
+    public function __construct(
+        string $token = ExceptionTokens::ENTITY_NOT_FOUND,
+        string $message = ExceptionTokens::ENTITY_NOT_FOUND,
```

Comment:
It should be used in every exception. The `$token` is what is used as a translation key. If you don't have a specific enough `$token`, then the translation of the error to the user won't make any sense. So while it's technically optional for a subclass to override it, it's required in the base exceptions constructors.

Generally the subclass does only receive an optional `$message`. The question is do we send to `\Exception::__construct()` the `$message` or the `$token` for it's own `$message` parameter. One way makes sure that we do not display non-translatable `$message`s to the user (aka, send `$token` as `$message`). The other ensures compatibility with NewRelic, Symfony's debug output, console exception output, etc.

I'm leaning towards sending `$message` to `\Exception::__construct()` and keeping `$token` internally because of all of the compatibility issues. We just have to make sure no user-facing application sends the `$message` instead of the `$token`.